Dojo example update with Dojo 1.10, fixes #1088.

asudoh opened the pull request:
This change bumps up Dojo version to 1.10, and does the following:

* Optimize the code to simplify the example
* Add unit tests (By [Intern](http://theintern.io))
* Make the code JSHint/JSCS-clean

sindresorhus commented on examples/dojo/bower.json:
```diff
@@ -1,7 +1,13 @@
-  "dependencies": {
-    "todomvc-common": "~0.3.0"
-  }
+	"name": "faketodmvc",
```
> todomvc-dojo

sindresorhus commented on examples/dojo/bower.json:
```diff
@@ -1,7 +1,13 @@
-    "todomvc-common": "~0.3.0"
-  }
+	"name": "faketodmvc",
+	"version": "0.0.1",
```
remove

sindresorhus commented on examples/dojo/bower.json:
```diff
@@ -1,7 +1,13 @@
-  }
+	"name": "faketodmvc",
+	"version": "0.0.1",
+	"dependencies": {
```
add `"private": true,`

sindresorhus commented on examples/dojo/bower.json:
```diff
@@ -1,7 +1,13 @@
+		"dijit": "~1.10.0",
+		"dojox": "~1.10.0",
+		"util": "dojo/util#~1.10.0"
+	}
```
add `"private": true,`

sindresorhus commented on examples/dojo/css/hidden.css:
```diff
@@ -0,0 +1,3 @@
+[hidden="true"] {
+	display: none;
+}
```
Should be inline in the HTML for to prevent flash of element hiding on load.

sindresorhus commented on examples/dojo/index.html:
```diff
@@ -4,29 +4,66 @@
-				mvc: { debugBindings: true }
-			};
+		<script type="dojo/require">
+			at: "dojox/mvc/at"
```
single-quotes

sindresorhus commented on examples/dojo/index.html:
```diff
@@ -4,29 +4,66 @@
+			</script>
+		</section>
+		<script type="text/javascript" src="js/main.js"></script>
+		<script type="text/javascript" src="js/lib/dojo/dojo.js"></script>
```
remove `type` attributes

sindresorhus commented on examples/dojo/js/todo/computed.js:
```diff
@@ -0,0 +1,161 @@
+	}
+
+	function removeHandles(handles) {
+		for (var h = null; (h = handles.shift());) {
```
while loop is probably better here.

sindresorhus commented on examples/dojo/js/todo/computed.js:
```diff
@@ -0,0 +1,161 @@
+			}
+		};
+	};
+});
```
Possible to use a dojo plugin of some kind instead? Feels weird having manually implement something like this.

asudoh replied:
Good point, however, unfortunately Dojo does not have a component for that purpose. Further discussion can be found at: http://dojo-toolkit.33424.n3.nabble.com/Binding-to-a-Stateful-s-computed-property-td3990310.html

sindresorhus replied:
That is a very old discussion though. I just feel like having to manually reinvent something like that is a big Dojo turn-off for anyone looking at this app.

sindresorhus replied:
Can you open a Dojo ticket for this too?

asudoh replied:
Agreed that it's something that should come from Dojo out-of-the-box. Created a ticket: https://bugs.dojotoolkit.org/ticket/18440

sindresorhus replied:
Can you add a TODO comment about removing it when the above is fixed an link to the ticket? So we don't forget.

asudoh replied:
Sure, done: https://github.com/asudoh/todomvc/blob/dojo110/examples/dojo/js/todo/computed.js#L1

sindresorhus commented on examples/dojo/js/todo/empty.js:
```diff
@@ -0,0 +1 @@
+define(function () {});
```
Why?

asudoh replied:
For: https://github.com/asudoh/todomvc/blob/dojo110/examples/dojo/js/main.js#L29-L34 Will add a comment for more explanation.

asudoh replied:
Added a comment to: https://github.com/asudoh/todomvc/blob/dojo110/examples/dojo/js/todo/empty.js#L1

sindresorhus commented on examples/dojo/js/todo/store/TodoLocalStorage.js:
```diff
@@ -0,0 +1,156 @@
+			return QueryResults(this.queryEngine(query, options)(this._loadStorage()));
+		}
+	});
+});
```
Same here. Would be better to use a reusable module. Or turn this into one if none exists.

asudoh replied:
There is one using IndexedDB [dojox/store/LocalDB](https://github.com/dojo/dojox/blob/master/store/LocalDB.js) but not one with localStorage (as you guessed). Created a ticket for that: https://bugs.dojotoolkit.org/ticket/18439

sindresorhus replied:
Can you add a TODO comment about removing it when the above is fixed an link to the ticket? So we don't forget.

asudoh replied:
Sure, done: https://github.com/asudoh/todomvc/blob/dojo110/examples/dojo/js/todo/store/TodoLocalStorage.js#L1

sindresorhus commented on examples/dojo/js/todo/widgets/Todo.js:
```diff
@@ -0,0 +1,104 @@
+	'./TodoEscape',
+	'./TodoFocus'
+], function (declare, lang, when, Deferred, Stateful,
+	registry, _WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin, _InlineTemplateMixin, at, computed) {
```
one function argument on each line for readability.

sindresorhus commented on examples/dojo/package.json:
```diff
@@ -0,0 +1,11 @@
+{
+	"name": "faketodomvc",
+	"version": "0.0.1",
+	"dependencies": {},
```
you can drop all of the above props as they're optional.

sindresorhus commented:
Awesome!

Make sure the [test suite](https://github.com/tastejs/todomvc/tree/master/browser-tests) is passing.

asudoh commented:
Thanks @sindresorhus for your review. Addressed your comments (except ones with my comment) as well as made sure the Mocha test passes, and created a new commit: https://github.com/asudoh/todomvc/commit/4ebe529dbd49163bb0443fa5f15aeb976f196223

sindresorhus commented:
Thanks @asudoh :dancer:
